src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore && !victim.hidden && victim.health > 0) {
+      victim.health -= 50;
+
+      if (victim.health <= 0) {
+        Animal.removeDeadAnimals();
```

Comment:
The `removeDeadAnimals` method is correctly called to remove dead animals from the `alive` array. This ensures that the list of alive animals is always up-to-date.